pontoon/sync/core.py

```diff
@@ -20,6 +21,27 @@
 
 log = logging.getLogger(__name__)
 
+def run_only_one(func):
+    """
+    Decorator ensures that there's only one running task with given task_name.
+    """
+
+    task_name = "celery.{}".format(func.__name__)
```

Comment:
This bars us from having two separate tasks with the same name. Normally celery tasks are named after their containing module and the function name to avoid naming conflicts.

Since this is meant to wrap tasks anyway, you may want to just decorate the wrapped function with Celery's task decorator (and pass arguments given to your decorator to it so that we can still pass arguments to celery), and set `bind` to True so that you can access `self.name`, which is a guaranteed unique task name generated by celery.